### ISSUE TITLE
ZeroDivisonError in inverse_sqrt scheduler

### ISSUE THREAD

**Issue by Sangh0** [huggingface/transformers#28835]
Hello! I've been findiing your repository very useful.

However, I've encountered a bug. It occurs when trying to train using the inverse_sqrt scheduler, leading to a ZeroDivisionError.
Please refer to the following code link for details.
Therefore, I suggest modifying the code from `decay = 1.0 / math.sqrt((current_step + shift) / timescale)` to `decay = 1.0 / math.sqrt((current_step + shift) / (timescale + 1e-9))`.

[Bug code](https://github.com/huggingface/transformers/blob/ec29d25d9f7109f3fdaadfa51515eb6745a136ba/src/transformers/optimization.py#L292)

![zerodivisionerror](https://github.com/huggingface/transformers/assets/47784418/5ce8c101-5da9-49c6-b8c4-2a79c9ab5f80)

@muellerzr @pacman100
Could you please check this issue?

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

```python
args = TrainingArguments(
    output_dir="./checkpoints",
    lr_scheduler_type="inverse_sqrt",  # ZeroDivisonError
)
```

### Expected behavior
```python
decay = 1.0 / math.sqrt((current_step + shift) / (timescale + 1e-9))
```

**Comment by Ericmagsllonstrigs** [comment]
@@

**Comment by Ericmagsllonstrigs** [comment]
@**![image](https://github.com/huggingface/transformers/assets/135127487/0584395c-7c48-440d-954e-72135b452fd1)**

**Comment by pacman100** [comment]
cc @muellerzr

**Comment by Sangh0** [comment]
Thank you for your interest in this issue. I will close this issue now.

**Comment by Ericmagsllonstrigs** [comment]
Yoissa
